cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -356,6 +356,19 @@ int VpiSignalObjHdl::set_signal_value(std::string &value)
     return set_signal_value(value_s);
 }
 
+int VpiSignalObjHdl::set_signal_value_str(std::string &value)
+{
+    s_vpi_value value_s;
+
+    std::vector<char> writable(value.begin(), value.end());
+    writable.push_back('\0');
+
+    value_s.value.str = &writable[0];
```

Comment:
nit: `value.c_str()`, which includes the trailing 0. Fine to leave to a follow-up PR.

Reply:
I was going to do that for both functions, but I get poor reception to changing non-pertinent code, and I am not making the implementations inconsistent. I will change both here though.

Reply:
Actually, there is a `const`ness issue with `c_str`. I am leaving it as it is.

Reply:
Good catch